packages/server/src/utils/HeaderMap.ts

```diff
@@ -1,4 +1,8 @@
 export class HeaderMap extends Map<string, string> {
+  // In order for TypeScript to prevent a standard `Map` from being compatible
+  // with a `HeaderMap`, we need some additional property on the class.
+  __identity = Symbol('HeaderMap');
```

Comment:
Can this be private? (Guessing no.)

Can this change be tested with a `@ts-expect-error`?

Reply:
(also re "can this be private", guessing this might even be a weird thing where if made private it sorta works within our ts project but then works differently when published as DTS.)